Electrs fixes

erikarvstedt opened the pull request:
The electrs package update is needed for fixing the exposed secret.

jonasnick commented on modules/electrs.nix:
```diff
@@ -5,11 +5,13 @@ let
   imports = [
     (mkRenamedOptionModule [ "services" "electrs" "nginxport" ] [ "services" "electrs" "TLSProxy" "port" ])
+    (mkRemovedOptionModule [ "services" "electrs" "high-memory" ] ''
+      This option has been removed because its implementation was defective, leading
```
I have never used this module outside of testing but I don't get what youre' saying here. If `high-memory` was set, then jsonrpc-import was _not_ set. The [electrs docs](https://github.com/romanz/electrs/blob/master/doc/usage.md#usage) state that if these options are not set in a 2GB memory system, then there will be `memory allocation of xxxxxxxx bytes failedAborted`. In @nixbitcoin's case this happened with 4GB memory even.

erikarvstedt replied:
I completely misread that if-statement. Sorry, that's embarrassing. Reverted.

jonasnick commented on pkgs/electrs/default.nix:
```diff
@@ -1,2933 +1,25 @@
-        sha256 = "0q0ssgpj9x5a6fda83nhmfydy7a6c0wvxm0jhncsmjx8qp8gw91m";
-        authors = [
-          "The Rand Project Developers"
-          "The Rust Project Devel
```
Do you know what exactly is the input to the hash? There seemed to have been problems with fully pinning the dependencies before we used crate2nix (https://github.com/fort-nix/nix-bitcoin/issues/66 @jb55). Other than that I'd be very happy to simplify this by going back to buildRustPackage.

erikarvstedt replied:
You mean `cargoSha256`? It's the fixed output hash on the result of running `cargo vendor`. (Here's the [derivation source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/fetchCargoTarball.nix).)
 With the current state of `buildRustPackage` the crate2nix approach has no advantages when it comes to pinning. Both fetch the exact dependencies defined in cargo.lock. 

To quote @jb55, the reason for switching to crate2nix was

> only to find that the rust packages in whatever rust channel nixpkgs was using was
> out of date

crate2nix doesn't pin `rustPackages` (that is rustc, cargo, ...), so in what way did it help here?

erikarvstedt commented on modules/electrs.nix:
```diff
@@ -66,16 +79,18 @@ in {
+        {
+          echo "cookie = \"${config.services.bitcoind.rpcuser}:$(cat ${secretsDir}/bitcoin-rpcpassword)\"";
+          echo "verbose = ${toString logLevels.${cfg.logLevel}}";
+        } >> electrs.toml
```
Fixed. That was a leftover from testing. (Although it wouldn't matter here because it's writing to a temporary runtime dir.)

jonasnick replied:
Oh I see, it's not writing to the datadir.

erikarvstedt commented:
I've removed option `logLevel` because I don't like the impl complexity for such a trivial setting.
I've added `extraArgs` instead, and I'll add an upstream patch to electrs to allow overriding the verbosity via cmdline args.
ACK? I'll squash then.

jonasnick commented:
reACK

jb55 commented:
ACK, I believe I just misunderstood how nixpkgs did it. The only benefit of my PR was to have build-cached dependencies, but using crate2nix is probably more brittle than just letting cargo do the build.
